presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -725,7 +726,7 @@ public synchronized void finishInsertIntoExistingPartition(
             String tableName,
             List<String> partitionValues,
             Path currentLocation,
-            List<String> fileNames,
+            List<StagingFileInfo> stagingFileInfos,
```

Comment:
For `finishInsertIntoExistingPartition` we shouldn't be doing rename twice. It should simply accept the list of committed staging files.